### DISCUSSION TITLE
Feature request: Google Search Console integration

### DISCUSSION THREAD

**Discussion by surjithctly** [umami-software/umami#645] [Ideas]
It would be a nice addition if we could see what kind of searches are made in google to reach our website and what pages on our site are most popular in search results.

https://developers.google.com/webmaster-tools

**Comment by Amerlander** [comment]
That's something I would love to see. That way, other users could get insights inside the search metrics without having direct access to the search console.

**Comment by jgui1129** [comment]
Upvoting this feature +1

**Comment by eugene-karuna** [comment]
Upvoting this feature +1

**Comment by jasalt** [comment]
I can recommend trying out SerpBear "search engine position tracking" tool which has some level of Search Console integration. It is also a MIT licensed TypeScript server application and simple to run eg. as container alongside Umami.

- https://docs.serpbear.com/ 
- https://github.com/towfiqi/serpbear

**Comment by kolosek** [comment]
@jasalt Great recommendation

**Comment by kevinrenskers** [comment]
It's the one feature I miss most coming from Plausible.

**Comment by calebneuf** [comment]
This would be great

**Comment by nathanbraun** [comment]
Chiming in here to say I just moved from Plausible/was surprised this isn't included.

**Comment by boutterudy** [comment]
Hey guys 👋 

I came across your feature request and thought it was spot on, so I decided to open a PR with the feature: https://github.com/umami-software/umami/pull/4117 😄 🎉 

Please do let me know if you have any feedback 🙏 
In the meantime, I hope the feature lives up to your expectations! 🫶

**Reply by julianwitzel** [reply]
This is awesome! Do you have an ETA for release on this feature?

Also I came across this API [https://dataforseo.com/](https://dataforseo.com/) and was wondering if it could be helpful to get even more SEO data into our page reports.

**Reply by julianwitzel** [reply]
Also what would be awesome is adding a way to see our scores in LLMs. Similar to what Webflow is building right now with their new AEO feature.

In general I think what Umami still lacks is the SEO aspect which is not easy though. And since your feature is exactly contributing to that I just thought I might add this here as well :)
